gips/data/landsat/landsat.py

```diff
@@ -200,6 +223,33 @@ def __init__(self, filename):
                 }
             self.visbands = [col for col in smeta['colors'] if col[0:4] != "LWIR"]
             self.lwbands = [col for col in smeta['colors'] if col[0:4] == "LWIR"]
+        elif c1_match:
+            utils.verbose_out('C1 asset', 2)
+
+            self.tile = c1_match.group('path') + c1_match.group('row')
+            year = c1_match.group('acq_year')
+            month = c1_match.group('acq_month')
+            day = c1_match.group('acq_day')
+            self.date = datetime.strptime(year + month + day, "%Y%m%d")
+
+            self.asset = 'C1'
+            self.sensor = "L{}{}".format(c1_match.group('sensor'), int(c1_match.group('satellite')))
+
+            smeta = self._sensors[self.sensor]
+            self.meta = {}
+            for i, band in enumerate(smeta['colors']):
+                wvlen = smeta['bandlocs'][i]
+                self.meta[band] = {
+                    'bandnum': i + 1,
+                    'wvlen': wvlen,
+                    'wvlen1': wvlen - smeta['bandwidths'][i] / 2.0,
+                    'wvlen2': wvlen + smeta['bandwidths'][i] / 2.0,
+                    'E': smeta['E'][i],
+                    'K1': smeta['K1'][i],
+                    'K2': smeta['K2'][i],
+                }
+            self.visbands = [col for col in smeta['colors'] if col[0:4] != "LWIR"]
+            self.lwbands = [col for col in smeta['colors'] if col[0:4] == "LWIR"]
```

Comment:
Seems like some of the C1 and DN blocks could be DRYed out (`meta`, `visbands`, and `lwbands` seem common), and `self.version` is not set for `C1`.